ffi/framebuffer.lua

```diff
@@ -78,19 +78,15 @@ function fb:init()
     -- asking the framebuffer for orientation is error prone,
     -- so we do this simple heuristic (for now)
     self.screen_size = self:getSize()
-    if self.screen_size.w > self.screen_size.h then
-        self.native_rotation_mode = self.ORIENTATION_LANDSCAPE
+    if self.screen_size.w > self.screen_size.h and self.device:isAlwaysPortrait() then
         self.screen_size.w, self.screen_size.h = self.screen_size.h, self.screen_size.w
-        if self.device:isAlwaysPortrait() then
-            -- some framebuffers need to be rotated counter-clockwise (they start in landscape mode)
-            self.debug("enforcing portrait mode by doing an initial rotation")
-            self.bb:rotate(-90)
-            self.blitbuffer_rotation_mode = self.bb:getRotation()
-            self.native_rotation_mode = self.ORIENTATION_PORTRAIT
-        end
-    else
+        -- some framebuffers need to be rotated counter-clockwise (they start in landscape mode)
+        self.debug("enforcing portrait mode by doing an initial rotation")
+        self.bb:rotate(-90)
+        self.blitbuffer_rotation_mode = self.bb:getRotation()
         self.native_rotation_mode = self.ORIENTATION_PORTRAIT
```

Comment:
That one line can go poof, since it's already set to this outside the branch.

Reply:
I'd also replace the debug message with a sterner warning to device maintainers that this should be fixed properly to avoid the rotated blitbuffer blitting penalty.

e.g.,

```
io.write("FB: Enforcing portrait mode by doing an initial BB rotation")                                                                                                                                                                                                                                                                                         
         io.flush()                                                                                                                                                                                                                                                                                                                                                      
         self.debug("FB: This prevents the use of blitting optimizations. This should instead be fixed on the device's side on startup.") 
```

(The first one is public, matching the framebuffer_linux logging, because it's good to know in bug reports. The other one is debug only for device maintainers).